{"path":"backwards/src/java/org/apache/lucene/index/IndexWriter#handleMergeException(Throwable,MergePolicy.OneMerge).mjava","commits":[{"id":"480d01e5b0ef8efb136d51670fec297ae5ae2c9c","date":1268821447,"type":0,"author":"Uwe Schindler","isMerge":false,"pathNew":"backwards/src/java/org/apache/lucene/index/IndexWriter#handleMergeException(Throwable,MergePolicy.OneMerge).mjava","pathOld":"/dev/null","sourceNew":"  final private void handleMergeException(Throwable t, MergePolicy.OneMerge merge) throws IOException {\n\n    if (infoStream != null) {\n      message(\"handleMergeException: merge=\" + merge.segString(directory) + \" exc=\" + t);\n    }\n\n    // Set the exception on the merge, so if\n    // optimize() is waiting on us it sees the root\n    // cause exception:\n    merge.setException(t);\n    addMergeException(merge);\n\n    if (t instanceof MergePolicy.MergeAbortedException) {\n      // We can ignore this exception (it happens when\n      // close(false) or rollback is called), unless the\n      // merge involves segments from external directories,\n      // in which case we must throw it so, for example, the\n      // rollbackTransaction code in addIndexes* is\n      // executed.\n      if (merge.isExternal)\n        throw (MergePolicy.MergeAbortedException) t;\n    } else if (t instanceof IOException)\n      throw (IOException) t;\n    else if (t instanceof RuntimeException)\n      throw (RuntimeException) t;\n    else if (t instanceof Error)\n      throw (Error) t;\n    else\n      // Should not get here\n      throw new RuntimeException(t);\n  }\n\n","sourceOld":null,"bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null},{"id":"9454a6510e2db155fb01faa5c049b06ece95fab9","date":1453508333,"type":5,"author":"Dawid Weiss","isMerge":false,"pathNew":"lucene/backwards/src/java/org/apache/lucene/index/IndexWriter#handleMergeException(Throwable,MergePolicy.OneMerge).mjava","pathOld":"backwards/src/java/org/apache/lucene/index/IndexWriter#handleMergeException(Throwable,MergePolicy.OneMerge).mjava","sourceNew":"  final private void handleMergeException(Throwable t, MergePolicy.OneMerge merge) throws IOException {\n\n    if (infoStream != null) {\n      message(\"handleMergeException: merge=\" + merge.segString(directory) + \" exc=\" + t);\n    }\n\n    // Set the exception on the merge, so if\n    // optimize() is waiting on us it sees the root\n    // cause exception:\n    merge.setException(t);\n    addMergeException(merge);\n\n    if (t instanceof MergePolicy.MergeAbortedException) {\n      // We can ignore this exception (it happens when\n      // close(false) or rollback is called), unless the\n      // merge involves segments from external directories,\n      // in which case we must throw it so, for example, the\n      // rollbackTransaction code in addIndexes* is\n      // executed.\n      if (merge.isExternal)\n        throw (MergePolicy.MergeAbortedException) t;\n    } else if (t instanceof IOException)\n      throw (IOException) t;\n    else if (t instanceof RuntimeException)\n      throw (RuntimeException) t;\n    else if (t instanceof Error)\n      throw (Error) t;\n    else\n      // Should not get here\n      throw new RuntimeException(t);\n  }\n\n","sourceOld":"  final private void handleMergeException(Throwable t, MergePolicy.OneMerge merge) throws IOException {\n\n    if (infoStream != null) {\n      message(\"handleMergeException: merge=\" + merge.segString(directory) + \" exc=\" + t);\n    }\n\n    // Set the exception on the merge, so if\n    // optimize() is waiting on us it sees the root\n    // cause exception:\n    merge.setException(t);\n    addMergeException(merge);\n\n    if (t instanceof MergePolicy.MergeAbortedException) {\n      // We can ignore this exception (it happens when\n      // close(false) or rollback is called), unless the\n      // merge involves segments from external directories,\n      // in which case we must throw it so, for example, the\n      // rollbackTransaction code in addIndexes* is\n      // executed.\n      if (merge.isExternal)\n        throw (MergePolicy.MergeAbortedException) t;\n    } else if (t instanceof IOException)\n      throw (IOException) t;\n    else if (t instanceof RuntimeException)\n      throw (RuntimeException) t;\n    else if (t instanceof Error)\n      throw (Error) t;\n    else\n      // Should not get here\n      throw new RuntimeException(t);\n  }\n\n","bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null}],"commit2Parents":{"480d01e5b0ef8efb136d51670fec297ae5ae2c9c":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":[],"cd5edd1f2b162a5cfa08efd17851a07373a96817":["9454a6510e2db155fb01faa5c049b06ece95fab9"],"9454a6510e2db155fb01faa5c049b06ece95fab9":["480d01e5b0ef8efb136d51670fec297ae5ae2c9c"]},"commit2Childs":{"480d01e5b0ef8efb136d51670fec297ae5ae2c9c":["9454a6510e2db155fb01faa5c049b06ece95fab9"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":["480d01e5b0ef8efb136d51670fec297ae5ae2c9c"],"cd5edd1f2b162a5cfa08efd17851a07373a96817":[],"9454a6510e2db155fb01faa5c049b06ece95fab9":["cd5edd1f2b162a5cfa08efd17851a07373a96817"]},"heads":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"roots":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"pathCommit":null}